server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -141,6 +144,27 @@ public static void download(TranslogTransferManager translogTransferManager, Pat
         }
     }
 
+    private static void deleteTranslogFilesNotUploaded(Path location, long uploadedGeneration) throws IOException {
+        // Delete translog files with generation > translogMetadata.getGeneration()
+        List<Long> generationsNotUploaded = Arrays.stream(FileSystemUtils.files(location))
```

Comment:
I think you can replace `Arrays.stream(FileSystemUtils.files(location))` with `Files.list(location)`

Reply:
I get `Stream<Path>' used without 'try'-with-resources statement` warning by using `Files.list()`.